dpnp/random/random_iface.py

```diff
@@ -0,0 +1,423 @@
+
+    Return random integers from `low` (inclusive) to `high` (exclusive).
+    Return random integers from the "discrete uniform" distribution of
+    the specified dtype in the "half-open" interval [`low`, `high`). If
+    `high` is None (the default), then results are from [0, `low`).
+
+    Parameters
+    ----------
+    low : int
+        Lowest (signed) integer to be drawn from the distribution (unless
+        ``high=None``, in which case this parameter is one above the
+        *highest* such integer).
+    high : int, optional
+        If provided, one above the largest (signed) integer to be drawn
+        from the distribution.
+    size : int or tuple of ints, optional
+        Output shape.  If the given shape is, e.g., ``(m, n, k)``, then
+        ``m * n * k`` samples are drawn.  Default is None, in which case a
+        single value is returned.
+    dtype : dtype, optional
+        Desired dtype of the result. Byteorder must be native.
+        The default value is int.
+    Returns
+    -------
+    out : array of random ints
+        `size`-shaped array of random integers from the appropriate
+        distribution, or a single such random int if `size` not provided.
+    See Also
+    --------
+    random_integers : similar to `randint`, only for the closed
+        interval [`low`, `high`], and 1 is the lowest value if `high` is
+        omitted.
+
+    """
+
+    if (use_origin_backend(low)):
+        return numpy.random.randint(low, high, size, dtype)
+
+    if size is None:
+        size = 1
+    elif isinstance(size, tuple):
+        for dim in size:
+            if not isinstance(dim, int):
+                checker_throw_value_error("randint", "type(dim)", type(dim), int)
+    elif not isinstance(size, int):
+        checker_throw_value_error("randint", "type(size)", type(size), int)
+
+    if high is None:
+        high = low
+        low = 0
+
+    low = int(low)
+    high = int(high)
+
+    if (low >= high):
+        checker_throw_value_error("randint", "low", low, high)
+
+    _dtype = numpy.dtype(dtype)
+
+    # TODO:
+    # supported only int32
+    # or just raise error when dtype != numpy.int32
+    if _dtype == numpy.int32 or _dtype == numpy.int64:
+        _dtype = numpy.int32
+    else:
+        raise TypeError('Unsupported dtype %r for randint' % dtype)
+
+    return dpnp_uniform(low, high, size, _dtype)
+
+
+def randn(d0, *dn):
+    """
+    If positive int_like arguments are provided, randn generates an array of shape (d0, d1, ..., dn),
+    filled with random floats sampled from a univariate “normal” (Gaussian) distribution of mean 0 and variance 1.
+
+    Parameters
+    ----------
+    d0, d1, …, dn : The dimensions of the returned array, must be non-negative.
+
+    Returns
+    -------
+    out : (d0, d1, ..., dn)-shaped array of floating-point samples from the standard normal distribution.
+
+    See Also
+    --------
+    standard_normal
+    normal
+
+    """
+
+    if (use_origin_backend(d0)):

```

Comment:
Could you please align overall approach in algorithm with other functions?
example:
https://github.com/IntelPython/dpnp/blob/fb5e82c850bade4e23457e19b1c7826e4c761019/dpnp/dpnp_iface_mathematical.py#L139-L151